CenterForOpenScience/osf.io
# [PLAT-432] Allow admin app to poke stuck registrations

## Purpose

It would be more convenient to have admins be able to identify and restart registrations using a nice UI, instead of getting Steve to run a script. This fix adds that UI.

## Changes

- Adds list view with list of stuck registrations with UI changes so its linked on the front page.
- Adds view for restarting regs via a button and modal
- Adds tests for both
- Adds a not strictly necessary line of code which add localhost to allowed hosts when the app is running locally.


## QA Notes

This is a tricky one to test because you have to fake a stuck registration and a lot of the staging data doesn't behave realistically and will resist restarting. The process I used to stick the registrations locally was just to kill the worker container, create the registration and start the worker then restart the registration via the Admin app.

 If you're doing this on staging you should be to stick a registration by registering a huge github project, however their is currently a fix in the works for this so that might not work. Registering Googledrive projects with the files that have the same name/path also may cause them to become stuck.

A stuck registration must be stuck for one day before it appears on the stuck registration list, so we may want to have someone go into to staging and mock the dates so we don't have to wait to test it.

Also using old staging data dumps seems you produce more stuck registrations then could be realistically possible. Keep an eye out for that.

## Documentation

This is an admin only feature and I don't think we have any admin only docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/projects/PLAT/issues/PLAT-432
